[Kartike] Duke Increments

KartikeKhandelwal opened the pull request.

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

j-lum committed:
Merge pull request #3 from j-lum/inline-javafx-tutorials

Inline JavaFX tutorials

j-lum committed:
Update Gradle tutorial to reflect new checkstyle config

j-lum committed:
Fix typo referring to `HelloWorld`

j-lum committed:
Add images to JavaFX tutorial 3

j-lum committed:
Merge pull request #7 from j-lum/tutorial-fix

Address issues #5 and #6

j-lum committed:
Remove references to DukeStub

j-lum committed:
Merge pull request #8 from j-lum/tutorial-fix

Remove references to DukeStub

j-lum committed:
Change file mode on `gradle` to be executable

j-lum committed:
Merge pull request #12 from j-lum/gradle+x

Change file mode on `gradle` to be executable (#9)

j-lum committed:
Merge pull request #12 from j-lum/gradle+x

Change file mode on `gradle` to be executable (#9)

jakq commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,18 @@
+                "] " + 
+                this.description + 
+                " (by: " + this.time +")"
+                );
```
do consider having less line breaks as it can get quite confusing for readers

jakq commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,51 @@
+        return("[" + this.type.charAt(0) + "][" + this.getStatusIcon() + "] " + this.description);
+    }
+
+    public boolean changedone(){
```
its good to break up/avoid toggle boolean methods if possible.

jakq commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,174 @@
+import java.io.*;
```
please remember that when importing files, list them instead of using the wildcard *. It helps when checking for errors. Also, avoid code where most of the method is in the client class and the main method.

kensonoenqh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,174 @@
+        System.out.println(line);
+
+        int count = 0;
+        int taskaddnum = 0;
```
Perhaps theres a better way to name this constant? There might be a name that means more while following coding standards.

kensonoenqh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,174 @@
+        int count = 0;
+        int taskaddnum = 0;
+        while(true){
+            String inputstring = sc.nextLine();
```
Consider inputString instead, to follow coding standards of camelCase variables.

kensonoenqh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,174 @@
+                        "__",
+                        "__");
+                    taskaddnum ++;
+                }
```
i like how you add single comments in each block, it makes the code easy to read and understand.

kensonoenqh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,174 @@
+
+    public static void output(String word){
+        System.out.println(line);
+        System.out.println("    "+word);
```
consider leaving whitespaces between the + sign.

kensonoenqh commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,16 @@
+    
+    String range;
+    public Event(String description, boolean isDone,String type,String range){
+        super(description,isDone,type);
```
consider use of whitespaces to make your code look cleaner.

kensonoenqh commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,51 @@
+    public String getdescreption(){
+        return this.description;
+    }
+    public String getLetterType(){
```
i think it is a good idea to have a method to retrieve the letter type.

kensonoenqh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,174 @@
+        int taskaddnum = 0;
+        while(true){
+            String inputstring = sc.nextLine();
+            if(inputstring.equals("bye")){
```
consider writing if else statements and try catch blocks in this form instead:
if (xx) {
    dosth;
} else {
    dosth;
}

kensonoenqh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,174 @@
+import java.io.*;
+import java.util.*;
+import java.util.Date;
```
consider leaving an extra line between your imports and start of code.

kensonoenqh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,174 @@
+    static String line = "    ____________________________________________________________";
+    public static LinkedList<Task> storage = new LinkedList<Task>(); 
     public static void main(String[] args) {
+        
```
You might not need to have an extra line here. Instead, consider putting a line before psvm.

kensonoenqh reviewed (commented):
Generally very nice code, only issue is that it could be slightly cleaner, with proper use of whitespaces and other coding standards.

damithc commented:
Please rectify the PR title to match the requirement. The PR author can edit the PR title.

damithc commented:
Please rectify the PR title to match the requirement. The PR author can edit the PR title.

damithc commented:
> Please rectify the PR title to match the requirement. The PR author can edit the PR title.

Reminder

damithc commented:
Another thing: don't commit binary files e.g., `*.class` files

damithc commented:
Tutor @andrewscw Please remind @troomtroom to correct the PR title during the next tutorial.

Commit:
new

Commit:
new

Commit:
commit2

Commit:
commit2

Commit:
commit3

Commit:
commit3

Commit:
commit3

Commit:
commit3

Commit:
commit4

Commit:
commit4

Commit:
commit4

Commit:
commit4

Commit:
Level-1

Commit:
Level-1

Commit:
Level-1

Commit:
Level-1

Commit:
added Level-2


Former-commit-id: fbdda9a5e2bee95a6c6f8162c15abaadc73000b4 [formerly 2c6dfcf0dd26ccb8ff3671984f40445b63b43ea9]
Former-commit-id: 8bbbb17f346fd27b2e472e74ffa72f8545256d3c

Commit:
added Level-2


Former-commit-id: 04c55bbee0eda614b4220ad09f7a9437e79ed7bf [formerly 6ccd85fa8bc85de5e37138802871d536957360a6]
Former-commit-id: 04c55bbee0eda614b4220ad09f7a9437e79ed7bf

Commit:
added Level-2


Former-commit-id: 04c55bbee0eda614b4220ad09f7a9437e79ed7bf [formerly a6a35d9fb62e7f22f36a5a469f0ba5ff8a5c7594 [formerly 04c55bbee0eda614b4220ad09f7a9437e79ed7bf] [formerly 04c55bbee0eda614b4220ad09f7a9437e79ed7bf [formerly 6ccd85fa8bc85de5e37138802871d536957360a6]]]
Former-commit-id: a6a35d9fb62e7f22f36a5a469f0ba5ff8a5c7594 [formerly 04c55bbee0eda614b4220ad09f7a9437e79ed7bf]
Former-commit-id: a6a35d9fb62e7f22f36a5a469f0ba5ff8a5c7594
Former-commit-id: 348136b13ba0e52e70f2a9b375e3aa576a7e2c6e

Commit:
added other gradle files

Commit:
added other gradle files

Commit:
other changes


Former-commit-id: a570df4abc9d0c061bb196dfb5ee11b61f7dd4e3
Former-commit-id: 0c1b36b63536b84241f5fee3d8847208666ecf3d

Commit:
other changes

Commit:
other changes

Commit:
new Level 2

Commit:
new Level 2

Commit:
new Level 2


Former-commit-id: 4e603a521fe3789dffcf1ac6468e68d48dd7ca0a
Former-commit-id: 45316c6d8a457ed73b15ba46045fd9a623a42346

Commit:
Level-4


Former-commit-id: d5c18866b9ca284c13de0e58b0d80714019b311f
Former-commit-id: f9117c5f9b1a828bcc2c13e86adf7debcbd900ae

Commit:
Level-4

Commit:
Level-4

Commit:
Added Level-5


Former-commit-id: 7ebc9408ac76b60c290f0b9cdc2dd66282188e92
Former-commit-id: 7a351ea50fa1d948c0e32d3e4e63d02252d636ee

Commit:
Added Level-5

Commit:
Added Level-5

Commit:
Added Level-5

Commit:
Added Level-5


Former-commit-id: bac98b83a9f3466330fa0c875a7411d44b6dbfda
Former-commit-id: ec6fafdeaaaa90b48801fd53f962ff0cd42b3441

Commit:
Added Level-5

Commit:
Add Level-6

Commit:
Add Level-6

Commit:
Add Level-6


Former-commit-id: 16ee6507fd93578efa57d82232dcbadc49fdddc0
Former-commit-id: 689bee2a492d417f37ca3aa7e32562815ee4b022

Commit:
Level-7

Commit:
Level-7


Former-commit-id: aab2ff712673714bac51c70d31b0ae1dbacf2b90
Former-commit-id: fe6586314fa7af41a436f6540f1456d89d798564

Commit:
Level-7

Commit:
Level-7

Commit:
Level-7

Commit:
Level-7


Former-commit-id: 7674f83acc70c7286d9bb6d69cbb86b45462351d
Former-commit-id: 384f30cb95f0927f6e5793c4a851c3e830cccd75

Commit:
Merge branch 'branch-Level-7'


Former-commit-id: 6e44135057fcc8162fa01e90c8806493c2778e44
Former-commit-id: 6748f7cb69d23a78fc7f681877573d65f4d4529f

Commit:
Merge branch 'branch-Level-7'

Commit:
Merge branch 'branch-Level-7'

Commit:
merge Level-7

Commit:
merge Level-7

Commit:
merge Level-7


Former-commit-id: a96d56cb8b870b4f0608f85b31b5e0c6cbaa3178
Former-commit-id: 7d77a1645370021072c44573808ee7b5092e4f78

Commit:
Level-8


Former-commit-id: 3fdaf7f728e2adfb98aa77a5d9b2f23f47c21812
Former-commit-id: 6aab5e903995436eff65afaf58f365b044dda77b

Commit:
Level-8

Commit:
Level-8

Commit:
Merge branch 'branch-Level-8'


Former-commit-id: c64d308d72ef42e70104795156f8d0d0c634dbcc
Former-commit-id: bd7eeb4dcdeccef3c74aa6ca5c8deb1780cc48d4

Commit:
Merge branch 'branch-Level-8'

Commit:
Merge branch 'branch-Level-8'

Commit:
A-MoreOOP

Commit:
A-MoreOOP

Commit:
A-MoreOOP


Former-commit-id: 1b72f19aa8ed54da74678532710bd58167f4bf84
Former-commit-id: 79c11e3a11fce2ef8d05139dab5233bea495657a

Commit:
add J-unit testing

Commit:
add J-unit testing


Former-commit-id: dc6f1e084d024cbce9a2771c805d6890ca2c5165
Former-commit-id: ad112fdad9dc64bbd30516d16d533b0c9ca717db

Commit:
add J-unit testing

Commit:
Add JavaDocs

Commit:
Add JavaDocs


Former-commit-id: 430c71aaa58ba6c89f1522c877b5e5aab45b1286
Former-commit-id: 9f6dfc9b19abf01d0126a105693c5bba340ddb33

Commit:
Add JavaDocs

Commit:
add A-JavaDOc

Commit:
add A-JavaDOc

Commit:
add A-JavaDOc


Former-commit-id: 0736258fa9374a8514b412561c215823ce40b53e
Former-commit-id: ecc5c36d6e487a3cab696ac2463de96d33e6d2d9

Commit:
Add coding standard

Commit:
Add coding standard

Commit:
Add coding standard


Former-commit-id: a9ecff56c39ffed0b59c80ee4cdc92e24d70ece3
Former-commit-id: e560dcb84b1cfe676b1e759daffe5f75545f2a8a

Commit:
add A-Gradle


Former-commit-id: d3907da6caff42f9c8520a51e230611185041375
Former-commit-id: c8e41925c57eda77632c1ff1284ddfc6caf78d86

Commit:
add A-Gradle

Commit:
add A-Gradle

KartikeKhandelwal committed:
Update gradle.yml

KartikeKhandelwal committed:
Update gradle.yml

KartikeKhandelwal committed:
Update gradle.yml

Former-commit-id: 063ac363987d27831c741ec87eaddbf827051be4
Former-commit-id: 5fe7b4a3da9bf8f8a60b9c272f104dd0558b4789

Commit:
Add Find Coommand


Former-commit-id: 43a33b2487d1e2e86eb479c80c110f8b5a9e00bd
Former-commit-id: 9e677b6cdceaa93681f92df87ba4278ca66502b8

Commit:
Add Find Coommand

Commit:
Add Find Coommand

Commit:
Merge branch 'master' of https://github.com/troomtroom/duke


Former-commit-id: 421ebfcc990d3082de5dc7d6459799ae82ab44ad
Former-commit-id: 3b81faa9b75145627fa6826dd777bd0f6c88899d

Commit:
Merge branch 'master' of https://github.com/troomtroom/duke

Commit:
Merge branch 'master' of https://github.com/troomtroom/duke

Commit:
Level-10

Commit:
Level-10

Commit:
Level-10


Former-commit-id: 755d79ea71bb0c4584b966b173feb5b7497d8e4a
Former-commit-id: 71d1f9a9f3da5cad2354485512cdcc57ebb6eec2

Commit:
add all features

Commit:
add all features


Former-commit-id: 683b40f77075c7420205f7c3c898daac01197e8b
Former-commit-id: 8a49f9ea677ce88accfb633d2da5fb33226376a8

Commit:
add all features

KartikeKhandelwal committed:
Set theme jekyll-theme-minimal

Former-commit-id: c6256d315acb8c9292d44d7ad11c80d123ddb0aa
Former-commit-id: b337f2f420e34072ec6c3f4f22e823de0b9831cf

KartikeKhandelwal committed:
Set theme jekyll-theme-minimal

KartikeKhandelwal committed:
Set theme jekyll-theme-minimal

Commit:
Final w/o GUI


Former-commit-id: fc02608ec00e9e17d1472eaaecb0a93c256cbcfa [formerly 7ddfc7541d41305272eb3955c8836ebb47650088]
Former-commit-id: 0a8bf1cb282eadb78933d5d8df888e51cc23ee36

Commit:
Merge branch 'master' of https://github.com/troomtroom/duke


Former-commit-id: 5194758e90eee7bc4390401da7c17f67a324d005 [formerly 13e8a3e46aa4348ccd9e9a12e78605937d5d699a]
Former-commit-id: 995b8d93bf6c16dfc2e9b05448af9848f79958a8

Commit:
Merge branch 'master' of https://github.com/troomtroom/duke


Former-commit-id: 793f9230c26f6a8bf5fa850e968efa2661426406

Commit:
add files

Commit:
add files

Commit:
correct files

Commit:
corrected files

Commit:
correct files

KartikeKhandelwal committed:
Set theme jekyll-theme-minimal

Commit:
Fix SaveFile issues


Former-commit-id: a37189140d8e9a46b8728ec75df9253beaa1c41d

Commit:
Fix SaveFile issues


Former-commit-id: a6af3c68150ec4293b3753161ba63e7c9e9af8a6

Commit:
Remove build


Former-commit-id: 8da946a3590ff89ee860729a0c06098f48f531a4

Commit:
Update to Docs and Website UG
